content/applications/inventory_and_mrp/barcode/setup/device_troubleshooting.rst

```diff
@@ -0,0 +1,212 @@
+==============================
+Barcode device troubleshooting
+==============================
+
+Odoo *Barcode* supports three main types of barcode scanners: USB scanners, bluetooth scanners, and
+mobile computer scanners. While configuring each type of scanner, common issues may arise, in which
+the scanners do not work as intended, and Odoo returns errors to the device.
+
+Read the sections below to identify general and unique device issues, related to specific, popular
+types of scanners.
+
+General issues
+==============
+
+Refer to the following sections below for common issues involving popular barcode scanner devices.
+
+For issues related to specific devices, refer to the :ref:`Android scanners
+<barcode/setup/android-scanners>` section for mobile computer scanners, or to the :ref:`Screenless
+scanners <barcode/setup/screenless-scanners>` section for USB and bluetooth scanners.
+
+Barcode cannot be read
+----------------------
+
+One common issue encountered when using barcode scanners is an error resulting from barcodes that
+cannot be read.
+
+This can occur due to any of the following reasons:
+
+- The barcode is damaged.
+- The device cannot read the required barcode type (some scanners can only read 2D barcodes).
+- The barcode being scanned is on a screen. Some scanners don't support this, and the barcodes
+  **must** be printed out to be scanned. This is most common with 1D barcodes.
+- The device has no battery, or is broken. To rule this out, follow the troubleshooting instructions
+  in the following sections.
+
+Odoo returns barcode error
+--------------------------
+
+All types of barcode scanners have their own device "language", which affects how they output
+barcode data to Odoo's *Barcode* app. Sometimes, this can cause Odoo *Barcode* to return a barcode
+error after scanning. This could be due to any of the following reasons:
+
+- The computer is configured with a different keyboard than the barcode scanner. To rule this out,
+  ensure that the device is configured with the same keyboard.
+
+  For example, if the computer is configured to use an FR-BE keyboard, configure the scanner to send
+  FR-BE keystrokes. The same logic applies if using a tablet instead of a computer.
+
+  For more information on configuring keystrokes, refer to the :doc:`Barcode scanner setup
+  <hardware>` documentation.
+- For mobile computer scanners (such as, Zebra devices), the scanner might interpret the barcode
```

Comment:
would remove oxford comma and specify that it's just an example
```suggestion
- For mobile computer scanners (such as Zebra devices, for example), the scanner might interpret the barcode
```

Reply:
Suggestion implemented ✅